### PR TITLE
Host docs to readthedocs

### LINKED ISSUE #10: Host readthedocs
Get docs on readthedocs (and make them prettier)

### PR DESCRIPTION
**Pull request by arjxn-py**
Hosts documentation on `readthedocs`.
Should fix #10

### CODE REVIEW

**Comment by arjxn-py** [comment]
Docs are building here : [https://pybamm-eis.readthedocs.io/en/latest/](https://pybamm-eis.readthedocs.io/en/latest/) 
Preetification remains

**Comment by rtimms** [comment]
awesome, thanks! just need to fix conflicts with the optional dependencies (literally merge #12 just as this opened)

let's merge this and worry about making it pretty later. can you also update the readme to point to the docs?

**Comment by arjxn-py** [comment]
> awesome, thanks! just need to fix conflicts with the optional dependencies (literally merged #12 just as this opened)
> 
> let's merge this and worry about making it pretty later. can you also update the readme to point to the docs?

Thanks for adding the tests @rtimms, so sorry that it took me more time than expected.
I've resolved the dependency conflicts, in addition to that we also have to note that currently `readthedocs` is building from my fork's [`arjxn-py:10-host-readthedocs`](https://github.com/arjxn-py/pybamm-eis/tree/10-host-readthedocs) branch so I'd update hosting too on readthedocs so that it builds from the main branch of upstream once this is merged (as `.readthedocs.yaml` was required to build so I just chose my fork for testing)

File: README.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # PyBaMM EIS
-PyBaMM EIS rapidly calculates the electrochemical impedance of any battery model defined using PyBaMM.
+
+PyBaMM EIS rapidly calculates the electrochemical impedance of any battery model defined using PyBaMM, see [the documentation](https://pybamm-eis.readthedocs.io/en/latest/)
```
Comment (arjxn-py) [inline README.md:3]: I've pointed to the docs here
I can also add this badge that is also pointing to the docs - 
[Documentation Status](https://pybamm-eis.readthedocs.io/en/latest/?badge=latest)

**Comment by rtimms** [comment]
great, thanks, @arjxn-py !

**Pull request merged**